integration-libs/punchout/root/services/punchout-detection.service.ts

```diff
@@ -24,4 +26,7 @@ export class PunchoutDetectionService {
       urlSections[0].includes(PUNCHOUT_SESSION_PAGE_URL)
     );
   }
+  isPunchoutSession(): boolean | undefined {
```

Comment:
we voluntary don't check if user is logged-in because it is used in auth-http-header service where user already has an invalid/expired token.